tutorials/dlp-hybrid-inspect/src/main/java/com/example/dlp/HybridInspectSql.java

```diff
@@ -63,104 +64,215 @@
+      System.out.println (String.format("Running scan of a data sources from external file [%s]",
+          cmd.getOptionValue(databasesOption.getOpt())));
+      LOG.info(String.format("> Retrieving list from (%s)",
+          cmd.getOptionValue(databasesOption.getOpt())));
+      try {
+        // create Gson instance
+        Gson gson = new Gson();
+
+        // create a reader
+        Reader reader = Files.newBufferedReader(Paths.get(cmd.getOptionValue(databasesOption.getOpt())));
+
+        // convert JSON file to map
+        List<Database> databases = new Gson().fromJson(reader, new TypeToken<List<Database>>() {}.getType());
+
+        // set up threading for the first level of threads (e.g. how many data sources do we scan in parallel)
+        final ExecutorService executorDB = Executors.newFixedThreadPool(new Integer(cmd.getOptionValue(threadPoolSizeOption.getOpt())).intValue());
+        final Map<String, Future<Integer>> futuresDB = new HashMap<>();
+
+        //Loop through all databases and inspect
+        for (Database tempDB : databases) {
+          LOG.info(String.format(tempDB.toString()));
+          Future<Integer> futureDB =
+              executorDB.submit(
+                  () -> inspectSQLDb(tempDB, cmd.getOptionValue(hybridJobNameOption.getOpt(),null), cmd.getOptionValue(hybridJobTriggerNameOption.getOpt(),null), runId));
+          futuresDB.put(tempDB.databaseName, futureDB);
+        }
+        executorDB.shutdown();
+
+        // close reader
+        reader.close();
+
+        for (String databaseName : futuresDB.keySet()) {
+          try {
+            // 10 minute timeout. Note that this a timeout on creating & sending the job request,
+            // not waiting for it to finish.  Larger data sources may need more time.
+            totalTableCount += futuresDB.get(databaseName).get(10, TimeUnit.MINUTES).intValue();
+            totalDatabaseCount++;
+          } catch (TimeoutException e) {
+            LOG.log(Level.WARNING, String.format("Timed out scanning table %s", databaseName), e);
+          } catch (Exception e) {
+            LOG.log(Level.WARNING, String.format("Exception scanning table %s", databaseName), e);
+          }
+        }
+
+      } catch (Exception ex) {
+        ex.printStackTrace();
+      }
     }
-    String databaseType = cmd.getOptionValue(sqlOption.getOpt());
-
-    inspectSQLDb(
-        cmd.getOptionValue(databaseInstanceServer.getOpt()),
-        cmd.getOptionValue(databaseInstanceServerDisplay.getOpt()),
-        cmd.getOptionValue(databaseName.getOpt()),
-        cmd.getOptionValue(tableName.getOpt()),
-        cmd.getOptionValue(databaseUser.getOpt()),
-        databasePassword,
-        databaseType,
-        Integer.parseInt(cmd.getOptionValue(sampleRowLimit.getOpt())),
-        cmd.getOptionValue(hybridJobName.getOpt()),
-        Integer.parseInt(cmd.getOptionValue(threadPoolSize.getOpt(), "4")));
+
+    java.util.Date endDate = new java.util.Date();
```

Comment:
Don't use `java.util.Date`. Use `java.time.Instant` and `java.time.Duration` instead.

```
Instant startTime = Instant.now();
// Do stuff
Instant endTime = Instant.now();
Duration elapsed = Duration.between(startTime, endTime);
long diffSeconds = elapsed.getSeconds();
```

Reply:
Updated